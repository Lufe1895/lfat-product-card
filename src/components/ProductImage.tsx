import React, { useContext } from 'react';
import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css';
import noImage from '../assets/no-image.jpg';

export interface Props {
    img?: string;
    className?: string;
    style?: React.CSSProperties 
}

export const ProductImage = ({ img, className, style }:Props) => {
    const { product } = useContext(ProductContext);
    let image:string = '';
    
    if (img) {
        image = img;
    } else if (product.img) {
        image = product.img;
    } else {
        image = noImage;
    }

    return (
        <img 
            className={ `${ styles.productImg } ${ className }` } 
            style={ style }
            src={ image } 
            alt="Product" 
        />
    );
}