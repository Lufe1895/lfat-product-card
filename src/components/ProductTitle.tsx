import React, { useContext } from 'react';
import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css';

export interface Props {
    className?: string 
    title?: string, 
    activeClass?: string;
    style?: React.CSSProperties 
}

export const ProductTitle = ({ className, style }:Props) => {
    const { product } = useContext(ProductContext);
    const { title } = product;
    return (
        <span 
            className={ `${ styles.productDescription } ${ className }` }
            style={ style }
        >
            { title }
        </span>
    );
}