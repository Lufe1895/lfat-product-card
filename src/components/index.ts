import { ProductTitle } from './ProductTitle';
import { ProductImage } from './ProductImage';
import { ProductButtons } from './ProductButtons';
import { ProductCard as ProductCardHOC } from '../components/ProductCard';
import { ProductCardHOCProps } from '../interfaces/interfaces';

export { ProductTitle } from './ProductTitle';
export { ProductImage } from './ProductImage';
export { ProductButtons } from './ProductButtons';

export const ProductCard:ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons,
});

export default ProductCard;