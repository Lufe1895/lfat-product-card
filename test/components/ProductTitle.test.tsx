import * as React from 'react';
import render from 'react-test-renderer';
import {} from '../../src/components';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductTitle } from '../../src/components/ProductTitle';
import { product } from '../../src/data/products';

describe('ProductTitle', () => {
    test('should mostrar el componente correctamente con el tÃ­tulo personalizado', () => {
        const wrapper = render.create(
            <ProductTitle title='Hola, mundo.' className='custom-class' />
        )

        expect(wrapper).toMatchSnapshot();
    });
    
    test('should mostrar el componente con el nombre del producto', () => {
        const wrapper = render.create(
            <ProductCard product={ product }>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect(wrapper).toMatchSnapshot();
    })
    
})
