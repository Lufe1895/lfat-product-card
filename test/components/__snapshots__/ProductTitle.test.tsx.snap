// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin Imagen
  </span>
</div>
`;

exports[`ProductTitle should mostrar el componente correctamente con el t√≠tulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  Hola, mundo.
</span>
`;
